src/core/server/plugins/plugins_system.test.ts

```diff
@@ -517,4 +546,48 @@ describe('start', () => {
     const log = logger.get.mock.results[0].value as jest.Mocked<Logger>;
     expect(log.info).toHaveBeenCalledWith(`Starting [2] plugins: [order-1,order-0]`);
   });
+
+  it('validates opensearch plugin installation', async () => {
+    [
+      createPlugin('order-1', { requiredOSPlugin: ['test-plugin'] }),
+      createPlugin('order-2'),
+    ].forEach((plugin, index) => {
+      jest.spyOn(plugin, 'setup').mockResolvedValue(`setup-as-${index}`);
+      jest.spyOn(plugin, 'start').mockResolvedValue(`started-as-${index}`);
+      pluginsSystem.addPlugin(plugin);
+    });
+
+    await pluginsSystem.setupPlugins(setupDeps);
+    await pluginsSystem.startPlugins(startDeps);
+    expect(opensearch.client.asInternalUser.cat.plugins).toHaveBeenCalledTimes(1);
+  });
+
+  it('validates opensearch plugin installation and errors out when plugin is not installed', async () => {
+    [
+      createPlugin('id-1', { requiredOSPlugin: ['missing-opensearch-dep'] }),
+      createPlugin('id-2'),
+    ].forEach((plugin, index) => {
+      jest.spyOn(plugin, 'setup').mockResolvedValue(`setup-as-${index}`);
+      jest.spyOn(plugin, 'start').mockResolvedValue(`started-as-${index}`);
+      pluginsSystem.addPlugin(plugin);
+    });
+
+    await pluginsSystem.setupPlugins(setupDeps);
+
+    await expect(pluginsSystem.startPlugins(startDeps)).rejects.toMatchInlineSnapshot(
+      `[Error: Plugin "id-1" has a dependency on OpenSearch plugin "missing-opensearch-dep" which is not installed on the cluster.]`
+    );
+  });
+
+  it('validates opensearch plugin installation and does not error out when there is no dependency', async () => {
```

Comment:
aren't we missing the happy path test case where there is a dependency, and that dependency is fulfilled?

Reply:
Do you mean the one in the line #550?

Reply:
yeah, I guess so. I don' think I read it as such due to the description and the fact that it doesn't include anything to validate success like `expect(pluginsStart).toBeInstanceOf(Map);`

Reply:
Updated